lib/asciidoctor/converter/semantic-html5.rb

```diff
@@ -53,10 +69,52 @@ def generate_document_title node
     end
   end
 
+  def generate_authors node
+    return if node.authors.empty?
+
+    if node.authors.length == 1
+      %(<p class="byline">
+#{format_author node, node.authors.first, 1}
+</p>)
+    else
+      result = ['<ul class="byline">']
+      node.authors.each_with_index do |author, index|
+        result << "<li>#{format_author node, author, index + 1}</li>"
+      end
+      result << '</ul>'
+      result.join LF
+    end
+  end
+
+  def format_author node, author, index
+    with_context 'author' do
+      %(<span class="author" data-index="#{index}">#{node.sub_replacements author.name}#{author.email ? %( #{node.sub_macros author.email}) : ''}</span>)
+    end
+  end
+
+  def with_context name, &block
+    (@convert_context ||= []).push name
+    result = block.call
+    @convert_context.pop
+    result
+  end
```

Comment:
Utility method to make the code a bit more readable.
I still don't know if it's better to use `&block` + `block.call`, implicit + `yield`, `Proc`, `lambda`... ?

Reply:
You should always use yield unless the situation calls for something different. For example, if you need to bind the block to another context, then you need something like instance_exec + block. But that's rare. I use blocks a lot in Asciidoctor PDF, so you can use that as a reference.